Projects/App/Sources/Journeys/TerminationFlowJourney.swift

```diff
@@ -3,26 +3,44 @@ import Presentation
 import hCore
 
 extension AppJourney {
-    static var terminationFlow: some JourneyPresentation {
+    static func terminationFlow(contractId: String, context: String) -> some JourneyPresentation {
         HostingJourney(
             ContractStore.self,
-            rootView: SetTerminationDate(),
+            rootView: SetTerminationDate(contractId: contractId, context: context),
             style: .modal
         ) {
             action in
-            if case .sendTermination = action {
-                sendTermination()
+            if case .sendTermination(let terminationDate, let context, let surveyURL) = action {
+                sendTermination(terminationDate: terminationDate, surveyURL: surveyURL)
+            } else if case .terminationFail = action {
+                terminationFail()
             }
         }
         .setScrollEdgeNavigationBarAppearanceToStandard
         .withDismissButton
     }
 
-    static func sendTermination() -> some JourneyPresentation {
+    static func sendTermination(terminationDate: String, surveyURL: String) -> some JourneyPresentation {
```

Comment:
Seems a bit weird that `terminationDate` is a `String` why is it not a `Date`?

Reply:
It's because I have to transform it to a FlowTerminationDateInput(input String) later as input to backend. So I havce to transform the date to a string later anyway but I could send it as a Date in this step yes